internal/lock.go

```diff
@@ -182,17 +182,19 @@ func (l *Lock) Save() error {
 	if err != nil {
 		return err
 	}
-	file, err := os.Create(l.path)
-	if err != nil {
+
+	dir := filepath.Dir(l.path)
+	tmpFile := filepath.Join(dir, fmt.Sprintf(".%s.tmp", filepath.Base(l.path)))
+
+	if err := os.WriteFile(tmpFile, res, 0644); err != nil {
 		return err
 	}
-	defer file.Close()
-	w := bufio.NewWriter(file)
-	_, err = w.Write(res)
-	if err != nil {
+
+	if err := os.Rename(tmpFile, l.path); err != nil {
+		os.Remove(tmpFile) // Clean up temp file if rename fails
 		return err
 	}
-	w.Flush()
+
```

Comment:
You've changed the semantics of this function quite significantly.  Can you explain why this was needed?